test/provider/model/EventPump.unittest.ts

```diff
@@ -0,0 +1,147 @@
+import 'jest';
+import 'reflect-metadata';
+
+import {EventPump} from '../../../src/provider/model/EventPump';
+import {NotificationClosedEvent, NotificationActionEvent} from '../../../src/client';
```

Comment:
Are these types really only available at the top-level of the client? I know with the mocks it will probably work, but it think it's sensible to continue avoiding any code imports from client files with actual code in them into tests as we've been doing up until now.

Reply:
These types are in `client/indexts`, and aren't rexported anywhere. Think we want to pull them out elsewhere?

Reply:
Are they not available from `internal.ts` somewhere? Do we not use them in the provider somewhere?

Reply:
`internal.ts` imports them itself but doesn't re-export. Everywhere they're used they're imported from `client/index.ts`, including in the provider

Reply:
Hmm, strange. That seems backwards to me, but I'll have a look myself and report back in a bit.